packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -130,6 +130,74 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
     return super.validateCommand();
   }
 
+  // Parses Contents.json into a map, with the key to be the combination of (idiom, size, scale), and value to be the icon image file name.
+  Map<String, String> _parseIconContentsJson(String contentsJsonDirName) {
+    final Directory contentsJsonDirectory = globals.fs.directory(contentsJsonDirName);
+    if (!contentsJsonDirectory.existsSync()) {
+      return <String, String>{};
+    }
+    final File contentsJsonFile = contentsJsonDirectory.childFile('Contents.json');
+    final Map<String, dynamic> content = json.decode(contentsJsonFile.readAsStringSync()) as Map<String, dynamic>;
+    final List<dynamic> images = content['images'] as List<dynamic>? ?? <dynamic>[];
+
+    final Map<String, String> iconInfo = <String, String>{};
+
+    for (final dynamic image in images) {
+      final Map<String, dynamic> imageMap = image as Map<String, dynamic>;
+      final String? idiom = imageMap['idiom'] as String?;
+      final String? size = imageMap['size'] as String?;
+      final String? scale = imageMap['scale'] as String?;
+      final String? fileName = imageMap['filename'] as String?;
+
+      if (size != null && idiom != null && scale != null && fileName != null) {
+        iconInfo['$idiom $size $scale'] = fileName;
+      }
+    }
+
+    return iconInfo;
+  }
+
+  Future<void> _validateIconsAfterArchive() async {
+    final BuildableIOSApp app = await buildableIOSApp;
+    final String templateIconImageDirName = await app.templateAppIconDirNameForImages;
+
+    final Map<String, String> templateIconMap = _parseIconContentsJson(app.templateAppIconDirNameForContentsJson);
+    final Map<String, String> projectIconMap = _parseIconContentsJson(app.projectAppIconDirName);
+    // find project icons that are in conflict with template icons
+    final List<MapEntry<String, String>> conflictIcons = projectIconMap.entries
```

Comment:
`List<MapEntry<String, String>>` is kind of odd, can it be a dictionary instead?  The keys should be unique, right?

Reply:
the reason is i want to do a filter (don't think `map` has a filter api). i could convert the list back to Map but i don't think it's useful/necessary.